tests/tdastro/astro_utils/test_passbands.py

```diff
@@ -0,0 +1,219 @@
+        for band_id in band_ids:
+            assert band_id in passbands.normalized_system_response_tables  # band_id is in the dictionary
+            assert passbands.normalized_system_response_tables[band_id].shape == (3, 2)  # 3 rows, 2 columns
+            assert np.allclose(passbands.normalized_system_response_tables[band_id][:, 1], 0.5)  # vals = 0.5
+
+
+def test_passbands_get_in_band_flux():
+    """Test the calculation of in-band flux for given flux and normalized system response table."""
+    passbands = Passbands(bands=["test-band"])
+
+    # Mock transmission table data for test-band
+    normalized_system_response_table = np.array([[100.0, 0.5], [200.0, 0.75], [300.0, 0.25]])
+    flux = np.array([1.0, 2.0, 3.0])
+
+    expected_in_band_flux = np.trapz(
+        flux * normalized_system_response_table[:, 1], x=normalized_system_response_table[:, 0]
+    )
+
+    # Calculate in-band flux using the method
+    calculated_in_band_flux = passbands._get_in_band_flux(flux, normalized_system_response_table)
+
+    assert np.isclose(calculated_in_band_flux, expected_in_band_flux, rtol=1e-9, atol=1e-9)
+
+    # TODO maybe a different check - is this a place where we could try the color thing?
+
+
+def test_passbands_get_all_in_band_fluxes():
+    """Test that we can calculate the in-band fluxes for all bands given a PhysicalModel and times."""
+
+    class MockPhysicalModel:
+        """A mock PhysicalModel for testing purposes."""
+
+        def evaluate(self, times, wavelengths):
+            """Mock evaluate method returning a 2D array of fluxes."""
+            flux_matrix = np.zeros((len(times), len(wavelengths)))
+            for i, time in enumerate(times):
+                flux_matrix[i] = np.sin(wavelengths * time)  # Some mock flux values
+            return flux_matrix
+
+    passbands = Passbands(bands=["test-band"])
+
+    # Mock the normalized system response table for test-band
+    normalized_system_response_table = np.array([[100.0, 0.5], [200.0, 0.75], [300.0, 0.25]])
+    passbands.normalized_system_response_tables["test-band"] = normalized_system_response_table
+
+    # Define some mock times
+    times = np.array([0.1, 0.2, 0.3])
+
+    # Create a mock PhysicalModel instance
+    # model = MockPhysicalModel()
+    model = SplineModel(
+        times,
+        [100.0, 200.0, 300.0],
+        np.array([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]),
+        time_degree=1,
+        wave_degree=1,
+    )
+
+    # Calculate expected flux matrix
+    expected_flux_matrix = np.zeros((len(times), len(passbands.bands)))
+    for i, time in enumerate(times):
+        # flux = np.sin(normalized_system_response_table[:, 0] * time)
+        flux = model.evaluate(np.array([time]), normalized_system_response_table[:, 0])
+        expected_flux = np.trapz(
+            flux * normalized_system_response_table[:, 1], x=normalized_system_response_table[:, 0]
+        )
```

Comment:
This is kinda a duplicate of the library code, I'm not sure if we can consider it as a fair test